src/Rings/special_ideals.jl

```diff
@@ -0,0 +1,74 @@
+export katsura
+
+function _get_katsura_exponent(n::Int, m::Int, l::Int)
+    e = zeros(Int, n + 1)
+    if abs(l) == abs(m - l)
+        e[abs(l) + 1] = 2
+    else
+        e[abs(l) + 1] = 1
+        e[abs(m - l) + 1] = 1
+    end
+
+    return e
+end
+
+@doc Markdown.doc"""
+    katsura(n::Int)
+
+Given a natural number `n` returns the Katsura ideal generated by\
+$u_m - \sum_{l=n}^n u_{l-m} u_l $, $ 1 - \sum_{l = -n}^n u_l $
+where $u_{-i} = u_i$, and $u_i = 0$ for $i > n$ and $m \in \{-n, \ldots, n\}$. \
+Also note that indices have been shifted to start from 1.
+
+# Example
+```jldoctest
+julia> katsura(3)
+ideal(x1 + 2*x2 + 2*x3 + 2*x4 - 1, x1^2 - x1 + 2*x2^2 + 2*x3^2 + 2*x4^2, 2*x1*x2 + 2*x2*x3 - x2 + 2*x3*x4, 2*x1*x3 + x2^2 + 2*x2*x4 - x3)
+```
+"""
+function katsura(n::Int)
+    R, _ = PolynomialRing(ZZ, n + 1)
+    return katsura(R)
+end
+
+@doc Markdown.doc"""
+    katsura(R::MPolyRing)
+
+Returns the Katsura ideal in the given Polynomial Ring R.
+
+julia> R, _ = QQ["x", "y", "z"]
+(Multivariate Polynomial Ring in x, y, z over Rational Field, fmpq_mpoly[x, y, z])
+
+julia> katsura(R)
+ideal(x + 2*y + 2*z - 1, x^2 - x + 2*y^2 + 2*z^2, 2*x*y + 2*y*z - y)
```

Comment:
This is missing code fences around the example code...

I'll submit a PR trying to clean thi sup